bot/constants.py

```diff
@@ -36,62 +36,96 @@
 PYTHON_PREFIX = "!"
 
 
-class Branding:
-    cycle_frequency = int(environ.get("CYCLE_FREQUENCY", 3))  # 0: never, 1: every day, 2: every other day, ...
+class EnvConfig(BaseSettings):
+    """Our default configuration for models that should load from .env files."""
+
+    class Config:
+        """Specify what .env files to load, and how to load them."""
+
+        env_file = ".env.server", ".env",
+        env_file_encoding = "utf-8"
+
+
+class _Branding(EnvConfig):
+    EnvConfig.Config.env_prefix = "branding_"
+
+    cycle_frequency = 3  # 0: never, 1: every day, 2: every other day, ...
+
+
+Branding = _Branding()
 
 
 class Cats:
     cats = ["ᓚᘏᗢ", "ᘡᘏᗢ", "🐈", "ᓕᘏᗢ", "ᓇᘏᗢ", "ᓂᘏᗢ", "ᘣᘏᗢ", "ᕦᘏᗢ", "ᕂᘏᗢ"]
 
 
-class Channels(NamedTuple):
+class _Channels(EnvConfig):
+    EnvConfig.Config.env_prefix = "channels_"
+
     algos_and_data_structs = 650401909852864553
     bot_commands = 267659945086812160
     community_meta = 267659945086812160
     organisation = 551789653284356126
     data_science_and_ai = 366673247892275221
-    devlog = int(environ.get("CHANNEL_DEVLOG", 622895325144940554))
+    devlog = 622895325144940554
     dev_contrib = 635950537262759947
     mod_meta = 775412552795947058
     mod_tools = 775413915391098921
     off_topic_0 = 291284109232308226
     off_topic_1 = 463035241142026251
     off_topic_2 = 463035268514185226
     python_help = 1035199133436354600
-    sir_lancebot_playground = int(environ.get("CHANNEL_COMMUNITY_BOT_COMMANDS", 607247579608121354))
+    sir_lancebot_playground = 607247579608121354
     voice_chat_0 = 412357430186344448
     voice_chat_1 = 799647045886541885
     staff_voice = 541638762007101470
-    reddit = int(environ.get("CHANNEL_REDDIT", 458224812528238616))
+    reddit = 458224812528238616
 
 
-class Categories(NamedTuple):
-    help_in_use = 696958401460043776
+Channels = _Channels()
+
+
+class _Categories(EnvConfig):
+    EnvConfig.Config.env_prefix = "categories_"
+
+    python_help_system = 691405807388196926
     development = 411199786025484308
     devprojects = 787641585624940544
     media = 799054581991997460
     staff = 364918151625965579
 
 
-codejam_categories_name = "Code Jam"  # Name of the codejam team categories
+Categories = _Categories()
+
+CODEJAM_CATEGORY_NAME = "Code Jam"  # Name of the codejam team categories
+
 
+class _Client(EnvConfig):
+    EnvConfig.Config.env_prefix = "client_"
 
-class Client(NamedTuple):
     name = "Sir Lancebot"
-    guild = int(environ.get("BOT_GUILD", 267624335836053506))
-    prefix = environ.get("PREFIX", ".")
-    token = environ.get("BOT_TOKEN")
-    debug = environ.get("BOT_DEBUG", "true").lower() == "true"
-    in_ci = environ.get("IN_CI", "false").lower() == "true"
-    github_bot_repo = "https://github.com/python-discord/sir-lancebot"
+    guild = 267624335836053506
+    prefix = "."
+    token = ""
```

Comment:
```suggestion
    token: str
```
Make this required to be passed in

Reply:
We could make them a pydantic's `SecretStr`, but it will require calling `get_secret_value` for all of the secrets that we have.
I don't mind having that, it's just that it might not be that obvious to call that method whenever we use secrets, but we can enforce it during reviews i guess.